### PR TITLE
Feature/accessible offline

### LINKED ISSUE #20: [Login] If there is no internet access, the application should be accessible if already logged.
As a user of the application i want to access the app even offline if I was already signed in.

### PR DESCRIPTION
**Pull request by JonathanDuvalV**
Closes #20 

This PR add the ability to access the app when the user doesn't have an active internet connection and was previously logged in to the app.

- Added NetworkingService that uses the Connectivity package from flutter
- When no connections are available, forces fromCacheOnly to true when possible
- Modified unit tests from repository affected by the NetworkingService
- Added new tests

### CODE REVIEW

**Comment by JonathanDuvalV** [comment]
Just realized I haven't fixed the tests related to the view models yet. I'll fix them soon.

File: lib/core/viewmodels/profile_viewmodel.dart [hunk 1]
```diff
@@ -70,20 +76,28 @@ class ProfileViewModel extends FutureViewModel<List<Program>> {
...
             .then((value) => _userRepository.getPrograms().catchError(onError))
-            .whenComplete(() => setBusyForObject(isLoadingEvents, false));
+            .whenComplete(() {
+          setBusyForObject(isLoadingEvents, false);
+          displayNoConnectionToast();
+        });
         return value;
       });
 
+  Future displayNoConnectionToast() async {
```
Comment (AlexandreMarkus) [inline lib/core/viewmodels/profile_viewmodel.dart:86]: Duplicate, maybe add this to Utils instead

File: lib/core/viewmodels/schedule_viewmodel.dart [hunk 2]
```diff
@@ -65,10 +71,17 @@ class ScheduleViewModel extends FutureViewModel<List<CourseActivity>> {
           }
         }).whenComplete(() {
           setBusyForObject(isLoadingEvents, false);
+          displayNoConnectionToast();
         });
         return value;
       });
 
+  Future displayNoConnectionToast() async {
```
Comment (AlexandreMarkus) [inline lib/core/viewmodels/schedule_viewmodel.dart:79]: Duplicate here too

File: lib/core/viewmodels/startup_viewmodel.dart [hunk 3]
```diff
@@ -1,4 +1,5 @@
 // FLUTTER / DART / THIRD-PARTIES
+import 'package:notredame/core/services/networking_service.dart';
```
Comment (AlexandreMarkus) [inline lib/core/viewmodels/startup_viewmodel.dart:2]: Add to section Services

**Review by AlexandreMarkus — COMMENTED** [review]
Not tested yet

**Comment by JonathanDuvalV** [comment]
> Seems good to me!
> 
> I would add the connectivity type (wifi, mobile, none) inside createGithubIssue() under `"- **Version:** ${packageInfo.version} \n"`
> 
> Like this : `"- **Connectivity:** ${...} \n"`

Done!

**Comment by JonathanDuvalV** [comment]
iOS doesn't compile anymore for me. I have the same issue on master. It looks like there's a problem with a Firebase dependency 

[!] CocoaPods could not find compatible versions for pod "Firebase/Analytics":
  In snapshot (Podfile.lock):
    Firebase/Analytics (= 7.3.0)

  In Podfile:
    firebase_analytics (from `.symlinks/plugins/firebase_analytics/ios`) was resolved to 8.0.2, which depends on
      Firebase/Analytics (= 7.11.0)

Specs satisfying the `Firebase/Analytics (= 7.3.0), Firebase/Analytics (= 7.11.0)` dependency were found, but they required a higher minimum deployment target.

**Comment by AlexandreMarkus** [comment]
Testé sur Android! Est-ce que IOS aussi?

**Comment by JonathanDuvalV** [comment]
> Testé sur Android! Est-ce que IOS aussi?

iOS a des problemes avec les dependencies de firebase 😅 . Je regardes ca soon!

**Comment by JonathanDuvalV** [comment]
@AlexandreMarkus Also works on iOS now!

File: test/helpers.dart [hunk 4]
```diff
@@ -5,6 +5,7 @@ import 'package:flutter_localizations/flutter_localizations.dart';
 import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:logger/logger.dart';
+import 'package:notredame/core/services/networking_service.dart';
```
Comment (camillebrulotte) [inline test/helpers.dart:8]: Wrong section :D

File: test/viewmodels/grades_viewmodel_test.dart [hunk 5]
```diff
@@ -19,11 +19,13 @@ import 'package:notredame/core/constants/router_paths.dart';
 // OTHER
 import '../helpers.dart';
 import '../mock/managers/course_repository_mock.dart';
+import '../mock/services/networking_service_mock.dart';
```
Comment (camillebrulotte) [inline test/viewmodels/grades_viewmodel_test.dart:22]: Wrong placeee hehe

**Review by camillebrulotte — CHANGES_REQUESTED** [review]
Nice job jojoo ! 2 p'tits changements à te sugérer :D

**Comment by JonathanDuvalV** [comment]
> Nice job jojoo ! 2 p'tits changements à te sugérer :D

Done 😄

**Comment by camillebrulotte** [comment]
LGTM ! Mais je suis incapable de tester sur IOS puisqu'il faut debug ! ):

**Review by AlexandreMarkus — APPROVED** [review]

**Review by camillebrulotte — APPROVED** [review]

**Pull request merged**